Projektarbeit III/Assets/Scripts/Player/PlayerController/WalkingState.cs

```diff
@@ -45,14 +54,28 @@ public void UpdateState()
         {
             if (controller.IsGrounded())
             {
-                Debug.LogError("Cannot dash while grounded");
+                Debug.Log("Player is grounded");
+                Debug.Log("Cannot dash while grounded");
             }
             else
             {   
             controller.ChangeState(new DashingState(controller));
             controller.movementEditor.hasDashed = true;
             }
         }
+
+        // Check for wall and ceiling collisions
+        if (controller.IsWalkingAgainstWall())
+        {
+            Debug.Log("Player is touching a wall and walking against it");
+            controller.ChangeState(new WallStickingState(controller));
+        }
+
+        if (controller.IsCeilinged())
+        {
+            Debug.Log("Player is touching a ceiling");
+            controller.ChangeState(new IdleState(controller));
+        }
```

Comment:
eeeeehhh this is dangerous. If the hitbox allows the player to move through a small passage but the raycast doesn't the player is stuck. Or am I having a logic error here?